gdnative-core/src/core_types/geom/transform2d.rs

```diff
@@ -445,3 +470,25 @@ godot_test!(
         test_transform2d_behavior_impl()
     }
 );
+
+#[test]
+fn test_transform2d_constructor() {
+    use std::f32::consts::PI;
+
+    let scale = Vector2::new(2.0, 0.5);
+    let rotation = PI / 4.0;
+    let origin = Vector2::new(250.0, 150.0);
+
+    let tr = Transform2D::from_scale_rotation_origin(scale, rotation, origin);
+
+    assert_eq!(tr.origin, origin);
+
+    let actual_local_right = tr.basis_xform(Vector2::RIGHT);
+    let expected_local_right = Vector2::RIGHT.rotated(-rotation) * scale;
+    assert!(
+        actual_local_right.is_equal_approx(expected_local_right),
+        "{:?} != {:?}",
+        actual_local_right,
+        expected_local_right
+    );
+}
```

Comment:
Should we not test `xform()` instead of `basis_xform()`? Currently this only checks that the parameter `origin` is assigned to the field `origin`, but not how a user would expect to use it...